source/autocomplete/autocomplete.h

```diff
@@ -505,6 +505,18 @@ struct Autocomplete
             }
         }
 
+        //We discard the GhostWords like de, le, la configured in synonyms as below
+        //de,
+        //le,
+        //la,
+        //For each synonyms.key found in strTemp if synonyms.value is empty delete the substring
+        for(const auto & it : synonyms){
+            if  ((boost::regex_search(strTemp,boost::regex("\\<" + it.first + "\\>")))
+                && (it.second.empty())){
```

Comment:
you should test first the emptiness before checking the regexp (but it is useless if you use a specific container ;) )